src/library/styles/__tests__/createStyledComponent.spec.js

```diff
@@ -0,0 +1,121 @@
+/* @flow */
+import React from 'react';
+import { mount } from 'enzyme';
+import { createStyledComponent } from '../../styles';
+
+const mountButton = (props = {}, styles = {}, options = {}) => {
+  const Button = createStyledComponent('button', styles, options);
+  return mount(<Button {...props} />);
+};
+
+describe('createStyledComponent', () => {
+  const wrapper = mountButton({ type: 'button', xyz: true });
+  const button = wrapper.childAt(0);
+
+  it('forwards valid DOM attributes', () => {
+    expect(button.props().type).toBeDefined();
+  });
+
+  it('filters invalid DOM attributes', () => {
+    expect(button.props().xyz).toBeUndefined();
+  });
+
+  describe('with withProps option', () => {
+    const wrapper = mountButton(
+      {},
+      {},
+      {
+        withProps: {
+          type: 'button'
+        }
+      }
+    );
+    const button = wrapper.childAt(0);
+
+    it('passes props to component', () => {
+      expect(button.props().type).toBe('button');
+    });
+  });
+
+  describe('with filterProps option', () => {
+    const wrapper = mountButton(
+      { filterMe: true },
+      {},
+      {
+        filterProps: ['filterMe']
+      }
+    );
+    const button = wrapper.childAt(0);
+
+    it('filters filterProps', () => {
+      expect(button.props().filterMe).toBeUndefined();
```

Comment:
I may be misunderstanding, but I don't understand how this assertion is meaningfully different from the one on line 20. Wouldn't you need to filter a valid DOM attribute for this to be useful? Or, put another way, is asserting on `.props()` an accurate representation of what's actually in the DOM?

Reply:
Agreed, it is unclear if the prop is being filtered via the `filterProps` setting or via the DOM tag.  It would work if it were styling a Button component and not a button tag.  Regardless, I updated this to filter a valid DOM attribute.